### PR TITLE
feat(net): add .NET abstractions, RemotingAction, RemotingReply

### PR DESCRIPTION
**Pull request by vobradovich**

### CODE REVIEW

File: net/src/Sails.Remoting/RemotingReply.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
```
[the remark this answers is not in the document]
Comment (vobradovich) [reply net/src/Sails.Remoting/RemotingReply.cs:4]: type `AsyncLazy`
but will ber removed after refactoring

File: net/src/Sails.Remoting.Abstractions/IActionBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using GasUnit = Substrate.NetApi.Model.Types.Primitive.U64;
+using ValueUnit = Substrate.NetApi.Model.Types.Primitive.U128;
+
+namespace Sails.Remoting.Abstractions;
+
+public interface IActionBuilder<TAction>
```
[the remark this answers is not in the document]
Comment (vobradovich) [reply net/src/Sails.Remoting.Abstractions/IActionBuilder.cs:6]: don't need it

File: net/src/Sails.Remoting.Abstractions/IRemotingListener.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /// <summary>
+    /// Listen to Gear events
+    /// </summary>
+    /// <param name="cancellationToken">Propagates notification that operations should be canceled. <see cref="CancellationToken"/> </param>
+    /// <returns></returns>
+    IAsyncEnumerable<byte[]> ListenAsync(CancellationToken cancellationToken);
+}
+
+public interface IRemotingListener<T>
```
[the remark this answers is not in the document]
Comment (vobradovich) [reply net/src/Sails.Remoting.Abstractions/IRemotingListener.cs:16]: yeah, `where T : IType, new()`

File: net/src/Sails.Remoting.Abstractions/IReply.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Substrate.NetApi.Model.Types;
+
+namespace Sails.Remoting.Abstractions;
+
+public interface IReply<T> where T : IType, new()
```
[the remark this answers is not in the document]
Comment (vobradovich) [reply net/src/Sails.Remoting.Abstractions/IReply.cs:7]: what for?) generated client exposes only intrefaces

**Review by DennisInSky — APPROVED** [review]

**Pull request merged**